src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Utilities/AsyncLazy`1.cs

```diff
@@ -23,6 +24,30 @@ public static AsyncLazy<T> Create<T>(T value)
         => new(value);
 }
 
+internal sealed class AsyncLazy<T> : AsyncLazy<
+    T,
+    (Func<CancellationToken, Task<T>> asynchronousComputeFunction, Func<CancellationToken, T>? synchronousComputeFunction)>
+{
+    public AsyncLazy(T value) : base(value)
+    {
+    }
+
+    public AsyncLazy(Func<CancellationToken, Task<T>> asynchronousComputeFunction)
+        : this(asynchronousComputeFunction, synchronousComputeFunction: null)
+    {
+    }
+
+    public AsyncLazy(
+        Func<CancellationToken, Task<T>> asynchronousComputeFunction,
+        Func<CancellationToken, T>? synchronousComputeFunction)
+        : base(
+            asynchronousComputeFunction: static (t, c) => t.asynchronousComputeFunction(c),
+            synchronousComputeFunction: synchronousComputeFunction is null ? null : static (t, c) => t.synchronousComputeFunction!(c),
```

Comment:
and here we ensure that the lambdas here are static.

Reply:
Compiler isn't smart enough to figure out the bang isn't necessary on "t.synchronousComputeFunction!(c)"?

Reply:
It is not.